src/workerd/io/worker-interface.capnp

```diff
@@ -157,6 +157,126 @@ struct Trace @0x8e8d911203762d34 {
   executionModel @25 :ExecutionModel;
   # the execution model of the worker being traced. Can be stateless for a regular worker,
   # durableObject for a DO worker or workflow for the upcoming Workflows feature.
+
+  # =====================================================================================
+  # Additional types for streaming tail workers
+
+  struct Attribute {
+    # An Attribute mark is used to add detail to a span over its lifetime.
+    # The Attribute struct can also be used to provide arbitrary additional
+    # properties for some other structs.
+    # Modeled after https://opentelemetry.io/docs/concepts/signals/traces/#attributes
+    struct Value {
+      inner :union {
+        text @0 :Text;
+        bool @1 :Bool;
+        int @2 :Int32;
+        float @3 :Float32;
+      }
+    }
+    name @0 :Text;
+    value @1 :List(Value);
```

Comment:
Having a list here is consistent with OTel, but at odds with what I'm working on in #3180 / out of scope for Automated Tracing as envisioned it so far (there's nothing that would need to add more than one tag value). Maybe useful for custom spans in the future.

Reply:
I think for now let's stick with the otel consistency and allow this to be multi-valued